apm-agent-core/src/main/java/co/elastic/apm/agent/context/LifecycleListener.java

```diff
@@ -47,6 +47,34 @@
      */
     void start(ElasticApmTracer tracer);
 
+    /**
+     * Callback for when {@link ElasticApmTracer#pause()} has been called.
+     * <p>
+     * Typically, this method is used to reduce overhead on the application to a minimum. This can be done by cleaning
+     * up resources like object pools, as well as by avoiding tracing-related overhead.
+     * </p>
+     * <p>
+     * Exceptions thrown from this method are caught and handled so that they don't prevent further cleanup actions.
+     * </p>
+     *
+     * @throws Exception When something goes wrong performing the cleanup.
+     */
+    void pause() throws Exception;
+
+    /**
+     * Callback for when {@link ElasticApmTracer#resume()} has been called.
+     * <p>
+     * Typically, used in order to revert the actions taken by the {@link LifecycleListener#pause()} method, allowing
+     * the agent to restore all tracing capabilities
+     * </p>
+     * <p>
+     * Exceptions thrown from this method are caught and handled so that they don't prevent further cleanup actions.
```

Comment:
same as above, + reword it so we don't mention any cleanup (as it's a resume operation).

Reply:
Same as above, reworded the `cleanup` to match resume, thanks for noticing!